app/src/main/java/com/lx/ltuddd/democaculator/utils/InfixPostfix.java

```diff
@@ -0,0 +1,99 @@
+package com.lx.ltuddd.democaculator.utils;
+
+import java.util.ArrayList;
+import java.util.Stack;
+
+/**
+ * Created by DaiPhongPC on 7/11/2018.
+ */
+
+public class InfixPostfix {
+    public static long stringToLong(String input) {
+        try {
+            long result = Long.parseLong(input);
+            return result;
+        } catch (NumberFormatException e) {
+            e.printStackTrace();
+            return 0;
+        }
+    }
+
+    public static boolean isNumber(String number) {
+        try {
+            Long.parseLong(number);
+            return true;
+        } catch (NumberFormatException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    public static int checkPriority(String operand) {
+        switch (operand) {
+            case "+":
+            case "-":
+                return 1;
+            case "x":
+            case "/":
+                return 2;
+            case "^":
+                return 3;
+        }
+        return -1;
+    }
+
+    public static ArrayList<String> infixToPostfix(ArrayList<String> arrValue) {
+        Stack<String> stack = new Stack<>();
+        ArrayList<String> arrConvert = new ArrayList<>();
+        for (String s : arrValue) {
```

Comment:
vòng for này bên trong if else khá nhiều. em có thể dùng if kết hợp vs return cho dễ nhìn. (cái này tạm thời a không bắt buộc)